nixos/modules/system/boot/unl0kr.nix

```diff
@@ -10,18 +16,65 @@ in
         Whether to enable the unl0kr on-screen keyboard in initrd to unlock LUKS.
       '';
     };
+
+    settings = lib.mkOption {
+      description = ''
+        Configuration for `unl0kr`.
+
+        See `unl0kr.conf(5)` for supported values.
+
+        Alternatively, visit `https://gitlab.com/postmarketOS/buffybox/-/blob/unl0kr-2.0.0/unl0kr.conf`
+      '';
+
+      example = lib.literalExpression ''
+        {
+            general.animations = true;
+            theme = {
+                  default = "pmos-dark";
+                  alternate = "pmos-light";
+            };
+        }
+      '';
+      default = { };
+      type = lib.types.submodule { freeformType = settingsFormat.type; };
+    };
   };
 
   config = lib.mkIf cfg.enable {
-    meta.maintainers = [];
+    meta.maintainers = with lib.maintainers; [ hustlerone ];
     assertions = [
       {
         assertion = cfg.enable -> config.boot.initrd.systemd.enable;
         message = "boot.initrd.unl0kr is only supported with boot.initrd.systemd.";
       }
+      {
+        assertion = !config.boot.plymouth.enable;
+        message = "unl0kr will not work properly if plymouth is enabled.";
+      }
+      {
+        assertion = !config.hardware.amdgpu.initrd.enable;
+        message = "unl0kr may not work properly if video drivers are loaded on stage 1.";
+      }
```

Comment:
This doesn't seem like something we should be asserting. It's going to depend on the specific hardware, isn't it?

Reply:
it does both on Steam Deck and desktop machine